actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -139,6 +141,34 @@ def negotiate_mime(order)
         order.include?(Mime::ALL) ? format : nil
       end
 
+      class VariantInquirer # :nodoc:
+        delegate :==, :each, :empty?, to: :variants
+
+        def initialize(variants = [])
+          @variants = variants
+        end
+
+        def any?(*candidates)
+          (variants & candidates).any?
+        end
+
+        def to_ary
+          variants
+        end
+
+        protected
+          attr_reader :variants
```

Comment:
It doesn't really seem that weird to me. It's an object that can be converted to a plain-old array. Anyway, what's the case where you would need access to the array? I tried to think of one and drew a blank.